frontend/console/src/features/command-pallete/CommandPalette.tsx

```diff
@@ -0,0 +1,96 @@
+import { Combobox, ComboboxInput, ComboboxOption, ComboboxOptions, Dialog, DialogBackdrop, DialogPanel } from '@headlessui/react'
+import { CellsIcon } from 'hugeicons-react'
+import { useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { useSchema } from '../../api/schema/use-schema'
+import { type PaletteItem, paletteItems } from './command-palette.utils'
+
+type CommandPaletteProps = {
+  isOpen: boolean
+  onClose: () => void
+}
+
+export const CommandPalette: React.FC<CommandPaletteProps> = ({ isOpen, onClose }) => {
+  const navigate = useNavigate()
+  const { data: schemaData } = useSchema()
+  const [query, setQuery] = useState('')
+  const [items, setItems] = useState<PaletteItem[]>([])
+
+  useEffect(() => {
+    if (schemaData) {
+      const newItems = paletteItems(schemaData)
+      setItems(newItems)
+    }
+  }, [schemaData])
+
+  const filteredItems = query === '' ? [] : items.filter((item) => item.title.toLowerCase().includes(query.toLowerCase()))
```

Comment:
Should this check the subtitle instead of the title, since someone might want to search by module name or by the full ref?

Reply:
I think I'll wait on this change because I want to add a real fuzzy search library to make this even better. Just wanted to get the basics in first before adding that new dep.

Reply:
awesome!